### PR TITLE
🪟 🎨 Improve table widths for new streams table

### PR DESCRIPTION
**Pull request by edmundito**
## What
Related to #21880 

This change updates the column sizing on the new streams table so that:

* Columns that don't need to be resized stay fixed
* Does not truncate the sync mode when it's set to `Incremental | Deduped + history`
* Does not create a horizontal scroll if the namespace and stream names are too long
* Increases min app width to prevent the table headers from looking bad but also a width that is no more than what users support

https://www.loom.com/share/130ea8649b9143c48eee9198343097e4

## How
Revisited how the table adapts to sizing. Some columns were explicitly made to have a fixed width. Here's the setup:

* Checkbox: Fixed
* Sync button: Fixed
* Namespace/Streamname: Variable
* Sync Mode: Fixed (and pill button now renders in such a way that gives more room to the right for Deduped + history)
* Cursor/Primary key: Variable
* Arrow: Fixed
* Namespace/Streamname: Variable

### CODE REVIEW

**Comment by krishnaglick** [comment]
Screenshot 2023-02-13 at 3 58 19 PM
On new table this header moves awkwardly up. Maybe a longer flex bases is necessary.

Screenshot 2023-02-13 at 3 59 56 PM
On the old table there's an issue w/ the row highlighting and scrolling. Does not happen on the new table since there's no streams table scroll bar.

File: airbyte-webapp/src/components/connection/CatalogTree/next/CatalogTreeTableRow.tsx [hunk 1]
```diff
@@ -55,18 +55,22 @@ export const CatalogTreeTableRow: React.FC<StreamHeaderProps> = ({
...
-      <CatalogTreeTableCell size="small">
+      <CatalogTreeTableCell size="fixed" className={styles.syncCell}>
         <Switch size="sm" checked={stream.config?.selected} onChange={onSelectStream} disabled={disabled} />
       </CatalogTreeTableCell>
-      {/* <Cell>{fieldCount}</Cell> */}
+      {/* <CatalogTreeTableCell size="fixed" className={styles.fieldsCell}>
+        <Text size="sm" className={styles.cellText}>
+          124567
+        </Text>
+      </CatalogTreeTableCell> */}
```
[the remark this answers is not in the document]
Comment (edmundito) [reply airbyte-webapp/src/components/connection/CatalogTree/next/CatalogTreeTableRow.tsx:73]: Yes, this is temp space for the column count that still needs to be implemented. Used to make sure that these table updates make room fro the column count.

File: airbyte-webapp/src/components/connection/CatalogTree/next/CatalogTreeTableHeader.tsx [hunk 2]
```diff
@@ -68,21 +63,25 @@ export const CatalogTreeTableHeader: React.FC = () => {
...
       </HeaderCell>
-      {/* <TextCell>
-        <FormattedMessage id="form.fields" />
-      </TextCell> */}
+      {/* <HeaderCell size="fixed" className={styles.fieldsCell}>
+        Fields
+        <InfoTooltip>
+          <FormattedMessage id="connectionForm.sourceNamespace.info" />
+        </InfoTooltip>
+      </HeaderCell> */}
```
[the remark this answers is not in the document]
Comment (edmundito) [reply airbyte-webapp/src/components/connection/CatalogTree/next/CatalogTreeTableHeader.tsx:84]: This is a placeholder for the column count. I will add a TODO to these commented out sections.

**Review by krishnaglick — APPROVED** [review]
LGTM, even with hack since old table getting deprecated.

**Review by dizel852 — COMMENTED** [review]
LGTM
But faced same issues like @krishnaglick mentioned
image

**Comment by edmundito** [comment]
While working with @dizel852 I noticed that we had a lot of problems specific to the craeate connection step, so I had to update the layout there to add the `<MainPageWithScroll>` component during the create connection step. I did not add it to the other steps due to complications with the documentation panel. This will make the create connection step consistent with the behavior in the replication page.

**Comment by krishnaglick** [comment]
Re-tested, things still LGTM!

**Review by krishnaglick — APPROVED** [review]

**Pull request merged**